src/vunnel/providers/alpine/parser.py

```diff
@@ -87,21 +87,18 @@ def _download(self):
                 os.makedirs(self.secdb_dir_path, exist_ok=True)
 
             self.logger.info("downloading alpine secdb metadata from: {}".format(self.metadata_url))
-            r = requests.get(self.metadata_url, timeout=self.download_timeout)
-            if r.status_code == 200:
-                try:
-                    self.logger.debug("HTML parsing secdb landing page content for links")
-                    parser = SecdbLandingParser()
-                    parser.feed(r.text)
-                    links = parser.links
-                except:
-                    self.logger.warning("unable to html parse secdb landing page content for links")
-
-                if not links:
-                    self.logger.debug("string parsing secdb landing page content for links")
-                    links = re.findall(self._link_finder_regex_, r.text)
-            else:
-                r.raise_for_status()
+            r = self._download_metadata_url()
+            try:
+                self.logger.debug("HTML parsing secdb landing page content for links")
+                parser = SecdbLandingParser()
+                parser.feed(r.text)
+                links = parser.links
+            except:
```

Comment:
this looks like it will swallow errors, so the retry with backoff wont attempt to retry this again

Reply:
I'm not sure I understand?  The retry is on `self._download_metadata_url()`

Reply:
And it does seem to work when I run it locally

Reply:
```
vunnel run alpine
[INFO ] running alpine provider
[INFO ] downloading alpine secdb metadata from: https://secdb.alpinelinux.org
[WARNING] <function Parser._download_metadata_url at 0x10435b5e0> failed. Retrying in 3 seconds (attempt 1 of 5)
[WARNING] <function Parser._download_metadata_url at 0x10435b5e0> failed. Retrying in 6 seconds (attempt 2 of 5)
[WARNING] <function Parser._download_metadata_url at 0x10435b5e0> failed. Retrying in 12 seconds (attempt 3 of 5)
[INFO ] Downloading secdb edge main
[INFO ] Downloading secdb edge community
[INFO ] Downloading secdb v3.10 main
[INFO ] Downloading secdb v3.10 community
```

Reply:
> The retry is on self._download_metadata_url()

🤦 my bad